homeassistant/components/ring.py

```diff
@@ -0,0 +1,63 @@
+"""
+Support for Ring Doorbell/Chimes.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/ring/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+import homeassistant.loader as loader
+
+from requests.exceptions import HTTPError, ConnectTimeout
+
+
+REQUIREMENTS = ['ring_doorbell==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by Ring.com"
+
+NOTIFICATION_ID = 'ring_notification'
+NOTIFICATION_TITLE = 'Ring Sensor Setup'
+
+DOMAIN = 'ring'
+DEFAULT_CACHEDB = 'ring_cache.pickle'
+DEFAULT_ENTITY_NAMESPACE = 'ring'
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+SCAN_INTERVAL = timedelta(seconds=5)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up Ring component."""
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+
+    persistent_notification = loader.get_component('persistent_notification')
+    try:
+        from ring_doorbell import Ring
+
+        ring = Ring(username, password)
+        if ring.is_connected:
```

Comment:
You want to return `False` if not connected, otherwise the component will be seen as set up correctly but the platforms will not be able to find the entry in `hass.data`